gradle-plugin-testing-junit/README.md

```diff
@@ -0,0 +1,606 @@
+    - [Build Directory](#build-directory)
+- [Executing Gradle Builds](#executing-gradle-builds)
+    - [Basic Execution](#basic-execution)
+    - [Environment Variables](#environment-variables)
+- [Assertions](#assertions)
+    - [Fluent Assertion API](#fluent-assertion-api)
+    - [Task Outcome Assertions](#task-outcome-assertions)
+    - [Output Assertions](#output-assertions)
+    - [File Assertions](#file-assertions)
+- [Error Prone Checks](#error-prone-checks)
+    - [Why We Ship Error Prone Checks](#why-we-ship-error-prone-checks)
+    - [Configuring Error Prone](#configuring-error-prone)
+
+## Quick Start
+
+Apply the plugin to your plugin project:
+
+```gradle
+apply plugin: 'com.palantir.gradle-plugin-testing'
+```
+
+This automatically adds the testing framework dependencies, configures your test tasks, and enables Error Prone checks that enforce best practices and enable automated migrations.
+
+### IDE Setup - Syntax Highlighting
+
+Add the [gradle-idea-language-injector](https://github.com/palantir/gradle-idea-language-injector) plugin to your root `build.gradle`:
+
+```gradle
+plugins {
+    id 'com.palantir.idea-language-injector' version '<version>'
+}
+```
+
+This plugin automatically generates IntelliJ IDEA language injection configurations, enabling syntax highlighting for string literals.
+
+### Write your first test
+
+You can use `@BeforeEach` for shared setup:
+
+```java
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.project.RootProject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.palantir.gradle.testing.assertion.GradlePluginTestAssertions.assertThat;
+
+@GradlePluginTests
+class MyPluginTest {
+    @BeforeEach
+    void setup(RootProject project) {
+        project.buildGradle().plugins().add("my-plugin");
+        project.buildGradle().append("""
+                myPlugin {
+                    enabled = true
+                }
+                """);
+    }
+    
+    @Test
+    void plugin_applies_successfully(GradleInvoker gradle, RootProject project) {
+        assertThat(gradle.withArgs("tasks").buildsSuccessfully())
+            .output()
+            .contains("myTask");
+    }
+}
+```
+
+## Core Concepts
+
+### The `@GradlePluginTests` Annotation
+
+Mark test classes with `@GradlePluginTests` to enable Gradle plugin testing. This automatically:
+- Creates isolated project directories for each test
+- Runs tests against multiple Gradle versions
+- Provides parameter injection for test infrastructure
+
+### Parameter Injection
+
+Tests can request these parameters in any order:
+
+- **`GradleInvoker`** - Executes Gradle builds
+- **`RootProject`** - Root project directory and files
+- **`SubProject`** - Named subprojects (name derived from parameter name)
+
+Parameters work in `@Test`, `@BeforeEach`, and `@AfterEach` methods.
```

Comment:
these Parameters can be used in the constructor for a test class or a @Test, @BeforeEach, @AfterEach, @BeforeAll, or @AfterAll